File: SwissO.Droid/OverviewAdapter.cs
using Android.App;
using Android.Views;
using Android.Widget;
using System;
using System.Collections.Generic;

namespace SwissO.Droid {

    class OverviewAdapter : BaseAdapter<Event> {

        private readonly Activity context;
        private readonly List<Event> events;
        private readonly AppManager appManager;

        private OverviewLayout selected;

        public OverviewAdapter(List<Event> events, Activity context, AppManager appManager) {
            this.events = events;
            this.context = context;
            this.appManager = appManager;
            selected = null;
        }
        public override Event this[int position] => events[position];

        public override int Count => events.Count;

        public override long GetItemId(int position) {
            return events[position].Id;
        }

        public override View GetView(int position, View convertView, ViewGroup container) {
            if (convertView == null) {
                convertView = LayoutInflater.From(context).Inflate(Resource.Layout.listitem_overview, container, false);
                ((OverviewLayout)convertView).SetExpandViewClick();
            }
            Event element = events[position];
            ((OverviewLayout)convertView).Init(element, appManager, this);
            return convertView;
        }

        public void ExpandViewClick(OverviewLayout v) {
            if (v == selected) {
                v.Collapse();
                selected = null;
            }
            else {
                if (selected != null) {
                    selected.Collapse();
                }
                v.Expand();
                selected = v;
            }
        }
    }
}

File: app/src/main/java/ch/swisso/OverviewAdapter.java
package ch.swisso;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

import java.util.ArrayList;

public class OverviewAdapter extends BaseAdapter {

    private final MainActivity act;
    private final ArrayList<Event> events;

    public OverviewAdapter(MainActivity act, ArrayList<Event> events){
        this.act = act;
        this.events = events;
    }

    @Override
    public int getCount() {
        return events.size();
    }

    @Override
    public Object getItem(int position) {
        return events.get(position);
    }

    @Override
    public long getItemId(int position) {
        return events.get(position).getId();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(act).inflate(R.layout.listitem_overview, parent, false);
        }
        ((OverviewLayout)convertView).init(events.get(position), act);
        return convertView;
    }
}
